### PR TITLE
Always lift buffers when importing frozen programs

### PR DESCRIPTION
**Pull request by j2kun**
Reproduction: https://gist.github.com/j2kun/c2539e1fd9a7de0d10e9513cff0e989c

In fx_importer.py, import_frozen_program only lifts buffers from `state_dict` to inlined constants in the else block of `if hasattr(prog, "constants"):`

It should always lift these buffers since the program is frozen.  I ran into this because I was exporting a torch model that had batch norms in it, and the batch norm parameters were left as (many many) function arguments when I wanted to freeze them from the values they got during training.

(Followed up from Discord)

### CODE REVIEW

File: test/python/fx_importer/basic_test.py [hunk 1]
```diff
@@ -460,3 +460,27 @@ def foo(x, y):
     m = fx.export_and_import(Basic(), x, y, func_name="test_stack_trace")
     mlir_asm = m.operation.get_asm(enable_debug_info=True)
     print(mlir_asm)
+
+
+@run
+# Asserting a single input argument here, since the frozen program should have
+# its BatchNorm parameters lifted from function arguments to inlined constants.
```
Comment (sahas3) [inline test/python/fx_importer/basic_test.py]:
```suggestion
# its BatchNorm buffers lifted from function arguments to inlined constants.
```

File: test/python/fx_importer/basic_test.py [hunk 2]
```diff
@@ -460,3 +460,27 @@ def foo(x, y):
...
+# CHECK-LABEL: test_import_frozen_exported_program_with_multiple_buffers
+# CHECK: func.func @main(%[[ARG0:[a-zA-Z0-9]+]]: !torch.vtensor<[2,10,20,20],f32>)
+# CHECK-COUNT-8: = torch.vtensor.literal(dense_resource<{{.*}}> : tensor<10xf32>)
```
Comment (sahas3) [inline test/python/fx_importer/basic_test.py:472]: nit: A comment like # 2 BN layers × 4 float tensors each (weight, bias, running_mean, running_var) would help future readers understand the magic number: 8.

**Review by sahas3 — APPROVED** [review]
LG.

The PyTorch version failure was just fixed in `main`. Can you rebase and rerun the CIs? Thanks!

File: test/python/fx_importer/basic_test.py (hunk 2 above)
Comment (j2kun) [reply test/python/fx_importer/basic_test.py:472]: Done.

**Comment by j2kun** [comment]
The failure is vs torch-nightly, not sure if that's blocking submission, as it doesn't seem related to the changes in this PR.

**Comment by sahas3** [comment]
All CIs need to be passing for merging the change. 

It looks like some test in `fx_importer` config is failing. I think the `stable` branch only runs `onnx` (not sure why) -- so the failure is a real issue that needs to be fixed. Does this not reproduce locally?

**Comment by j2kun** [comment]
> All CIs need to be passing for merging the change.
> 
> It looks like some test in `fx_importer` config is failing. I think the `stable` branch only runs `onnx` (not sure why) -- so the failure is a real issue that needs to be fixed. Does this not reproduce locally?

Thanks, that was helpful. Fixed in 5374cd3 (don't pass in buffers that are now automatically lifted)

**Comment by j2kun** [comment]
@sahas3 I felt the last two commits added are nontrivial enough that they should warrant a second review.

**Review by sahas3 — APPROVED** [review]
LG, the new edit makes sense to me.
